tests/end_to_end/features/autovariants.feature

```diff
@@ -0,0 +1,47 @@
+Feature: AutoVariants
+    Scenario: Same auto variant for same transformation 
+        Given Featureform is installed
+        When I turn on autovariants
+        And I create a "hosted" "insecure" client for "localhost:7878"
+        And I register "s3" filestore with bucket "featureform-spark-testing" and root path "data"
+        And I register databricks
+        And I register transactions_short.csv
+        And I register a transformation with auto variant
+        Then I should be able to reuse the same variant for the same transformation
+        And I can get the transformation as df
+        And I turn off autovariants
```

Comment:
may be a good idea to convert `I turn on/off autovariants` into @before & @after hooks to avoid test pollution if a scenario fails in a weird way. Those get defined in the step definitions files.